packages/components/src/utils/controllers/ElementStore.ts

```diff
@@ -180,6 +180,33 @@ export class ElementStore<TItem extends HTMLElement> implements ReactiveControll
     this.addItem(newItem, index);
   }
 
+  /**
+   * Implements binary search on the cache, to get the index of the nearest newItem position.
+   *
+   * @param cache - The array which contains the list of dom node elements.
+   * @param newItem - The new node element which is needed to be added in the list item.
+   * @returns
+   *  - The index of the existing element which is the nearest one to the new item.
+   *  - If the index is -1, then the item is added at the end of cache.
+   */
+  private getIndexToInsertInCache(newItem: TItem): number {
+    // If the new item is located at the last place, then we can check the last -1 cache item and return if valid
+    if (this.cache.length && !isAfter(this.cache.at(-1)!, newItem)) {
```

Comment:
we can save few more instructions if:
```suggestion
    if (!this.cache.length || !isAfter(this.cache.at(-1)!, newItem)) {
```